app/processing/payment/get-period-from-schedule.js

```diff
@@ -0,0 +1,10 @@
+const getPeriodFromSchedule = (totalValue, totalPayments, paymentValue, schedule, increment, unit) => {
+  const incrementValue = Math.trunc(totalValue / totalPayments)
+  const incrementsInPayment = Math.trunc((paymentValue / incrementValue) - 1)
+  const firstPeriodIndex = schedule.findIndex(x => x.outstanding)
+  const firstPayment = schedule[firstPeriodIndex]
+  const lastPayment = schedule[firstPeriodIndex + incrementsInPayment]
+  return `${firstPayment.dueDate.format('MMMM')} to ${lastPayment.dueDate.add(increment - 1, unit).format('MMMM YYYY')}`
```

Comment:
To note a post payment adjustment will cause problems here.  As it will with several other areas of the code because the payment request being used is only the completed one.  we need a mix of in progress and completed for post payment adjustments for the values to be correct.